samples/java/src/test/java/br/com/example/leadIntegration/LeadIntegrationApplicationTests.java

```diff
@@ -0,0 +1,91 @@
+package br.com.example.leadIntegration;
+
+import br.com.example.leadIntegration.models.Lead;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+
+import java.net.URI;
+import java.util.Base64;
+
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+class LeadIntegrationApplicationTests {
+
+	@LocalServerPort
+	private int port;
+
+	@Autowired
+	private TestRestTemplate restTemplate;
+
+	@Value("${spring.security.user.name}")
+	private String userName;
+
+	@Value("${spring.security.user.password}")
+	private String userPassword;
+
+	@Test
+	public void healthCheckShouldReturnOk() throws Exception {
+
+		final String baseUrl = "http://localhost:"+port+"/leads/health-check";
+		URI uri = new URI(baseUrl);
+
+		ResponseEntity<String> result = this.restTemplate.getForEntity(uri, String.class);
+
+		Assert.assertEquals(200, result.getStatusCodeValue());
+		Assert.assertEquals("{\"message\": \"OK\"}", result.getBody());
+
+	}
+
+	@Test
+	public void recieveLeadShouldBeSuccess() throws Exception {
+
+		final String baseUrl = "http://localhost:"+port+"/leads/lead";
+		URI uri = new URI(baseUrl);
+
+		Lead lead = new Lead("VivaReal", "2017-10-23T15:50:30.619Z", "59ee0fc6e4b043e1b2a6d863", "87027856",
```

Comment:
@giovannicaldeira for more fluent code ( in my opinion ) we can use the builder pattern, the lombok do this magic for us https://projectlombok.org/features/Builder

Reply:
@robsonoduarte added Builder annotation.

```java
Lead lead = Lead.builder()
				.leadOrigin("VivaReal")
				.timestamp("2017-10-23T15:50:30.619Z")
				.originLeadId("59ee0fc6e4b043e1b2a6d863")
				.originListingId("87027856")
				.clientListingId("a40171")
				.name("Nome Consumidor")
				.email("nome.consumidor@email.com")
				.ddd("11")
				.phone("999999999")
				.message("Olá, tenho interesse neste imóvel. Aguardo o contato. Obrigado.")
				.build();
```